src/rgw/rgw_sal_dbstore.cc

```diff
@@ -2029,12 +2030,14 @@ extern "C" {
 
       if (!dbsm ) {
```

Comment:
why do you check if `dbsm` is not null?
do you have an overloaded `new` operator?

same comment on line 2028.

Reply:
I could add assert() but I couldn't understand why coverity reports that we need to check if it is not null only if new operator is overloaded..Its the same case in other stores too - https://github.com/ceph/ceph/blob/main/src/rgw/rgw_sal_rados.cc#L3606

Reply:
this one is not a comment from Coverity, this is my comment :-)
throughout the ceph codebase, we *do not check* for failures of memory allocations. this is because the best thing you can do in such a case is to let the OS kill your process with OOM (Out Of Memory).
trying to handle OOM issues in the code is possible only in code bases where all memory is preallocated on startup and the new operators are overloaded to use this preallocated memory.
in systems such as ours, any attempts of this kind would probably just lead to more issues and would make it harder to debug the root cause.
note that "assert()" is compiled out in non-debug builds - so, it is not a good solution in this case.

Reply:
Got it. Thank you!